__tests__/app.test.js

```diff
@@ -111,4 +112,52 @@ describe("app", () => {
 
     })
 
+    describe("POST /api/articles/:article_id/comments", () => {
+        test("should return expected user and comment", () => {
+            return request(app).post("/api/articles/2/comments").send({
+                username: 'icellusedkars',
+                body: 'such a big fan wow'
+            }).expect(201).then((res) => {
+
+                const comment = res.body.comment
+
+                expect(comment.body).toBe("such a big fan wow");
+                expect(comment.author).toBe("icellusedkars");
+                expect(comment.article_id).toBe(2);
+                expect(comment.votes).toBe(0);
+
+                expect(comment).toMatchObject({
+                    comment_id: expect.any(Number),
+                    votes: expect.any(Number),
+                    created_at: expect.any(String),
+                    author: expect.any(String),
+                    body: expect.any(String),
+                    article_id: expect.any(Number),
+                });
+            })
+        })
+
+        test("expect 400 when missing username", ()=>{
+            return request(app).post("/api/articles/2/comments").send({
+                username: '',
+                body: 'such a big fan wow'
+            }).expect(400).then(({body}) => {
+                expect(body.msg).toBe('Bad Request')
+            })
+
+        })
+
+        test("expect 400 when missing body", ()=>{
+            return request(app).post("/api/articles/2/comments").send({
+                username: 'icellusedkars',
+                body: ''
+            }).expect(400).then(({body}) => {
+                expect(body.msg).toBe('Bad Request')
+            })
+
+        })
+
+        
+    })
```

Comment:
It is also possible that we could be missing the username or body properties altogether, we need a 'sad path' test for this.